src/modules/Dashboard/features/Applet/AddUser/AddUserForm/AddUserForm.tsx

```diff
@@ -61,132 +63,131 @@ export const AddUserForm = ({ getInvitationsHandler }: AddUserFormProps) => {
 
   const resetForm = () => reset();
 
-  const onSubmit = async (values: FormValues) => {
-    try {
-      if (id) {
-        const options = {
-          ...values,
-          accountName: values.accountName || authData?.account?.accountName || '',
-        };
-
-        await getAppletInvitationApi({ appletId: id, options });
-        await getInvitationsHandler();
+  const { error, execute } = useAsync(postAppletInvitationApi, async () => {
+    await getInvitationsHandler();
+    resetForm();
+  });
 
-        if (options.accountName && options.role !== Roles.User) {
-          dispatch(setAccountName({ accountName: options.accountName }));
-        }
-        setErrorMessage('');
-        resetForm();
-      }
-    } catch (e) {
-      setErrorMessage(getErrorMessage(e));
+  const onSubmit = (values: FormValues) => {
+    if (id) {
+      execute({
+        url: getUrl(values.role),
+        appletId: id,
+        options: values,
+      });
     }
   };
 
   const updateFields = (e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
     const { value } = e.target;
-    const { nickName, MRN, accountName, users } = Fields;
+    const { nickname, secretUserId, workspacePrefix, respondents } = Fields;
 
-    if (value === Roles.User) {
-      register(nickName, { value: '' });
-      register(MRN, { value: '' });
-      unregister(accountName);
-      unregister(users);
+    if (value === Roles.Respondent) {
+      register(nickname, { value: '' });
+      register(secretUserId, { value: '' });
+      unregister(workspacePrefix);
+      unregister(respondents);
     } else {
-      unregister(nickName);
-      unregister(MRN);
+      unregister(nickname);
+      unregister(secretUserId);
       if (value === Roles.Reviewer) {
-        register(users, { value: [] });
+        register(respondents, { value: [] });
       } else {
-        unregister(users);
+        unregister(respondents);
       }
-      if (accountNameShowed) {
-        register(accountName, { value: '' });
+      if (workspaceNameShowed) {
+        register(workspacePrefix, { value: '' });
       }
```

Comment:
may be it would be better to avoid nested conditions? like this
`if (value !== Roles.Respondent && value !== Roles.Reviewer) {
  unregister(nickname);
  unregister(secretUserId);
  unregister(respondents);
  workspaceNameShowed && register(workspacePrefix, { value: '' });
}

if (value === Roles.Respondent) {
  register(nickname, { value: '' });
  register(secretUserId, { value: '' });
  unregister(workspacePrefix);
  unregister(respondents);
}

if (value === Roles.Reviewer) {
  register(respondents, { value: [] });
}`

Reply:
fixed